### PR TITLE
[rpc] remove deprecated generate method

### PR DESCRIPTION
**Pull request by Sjors**
As announced in v0.18, the wallet generate rpc method is deprecated and will be fully removed in v0.19. 

Clients should transition to using the node rpc method `generatetoaddress`.

### CODE REVIEW

**Comment by laanwj** [comment]
Nice cleanup. utACK.

File: src/wallet/rpcwallet.cpp
Comment (jnewbery) [inline src/wallet/rpcwallet.cpp]: I expect you can also remove the `#include <rpc/mining.h>` include from this file.

File: test/functional/rpc_deprecated.py
Comment (jnewbery) [inline test/functional/rpc_deprecated.py]: Rather than change this file around, can you just revert it to how it was pre-commit c9f02955b2e9062808a9455c4ee7d52cf401eef5.

**Review by jnewbery — COMMENTED** [review]
This should be marked as WIP and not merged before the 0.18 branch split.

I agree that this should go in before #15492.

File: test/functional/rpc_deprecated.py
Comment (Sjors) [reply test/functional/rpc_deprecated.py]: Main reason for changing this file around is improving documentation; `generate` allows a much simpler example than `createmultisig`.

File: src/wallet/rpcwallet.cpp
Comment (Sjors) [reply src/wallet/rpcwallet.cpp]: Done

**Comment by Sjors** [comment]
Why WIP? That seems more appropriate for more experimental code. This PR is already tagged 0.19 so that should be clear enough.

**Comment by jnewbery** [comment]
Tagging 0.19 is fine. Just as long as the maintainers are aware that this shouldn't be merged prior to the branch split.

**Comment by ryanofsky** [comment]
This PR seems reasonable, but I would like to remove:

> Removal helps with separating the wallet from the node in #15288 

from the PR description. I don't think this is true in any significant sense. If there really is a need to relate this PR to #15288, you could say:

> Removal means that the `interfaces::Chain` class no longer needs to have a `generateBlocks` method added in #15288

I think there is some confusion in the #15288 discussion that `generateBlocks` method requires additional makefile changes that other `Chain` methods don't require, but this is not true. The only reason makefile changes are part of the `generateBlocks` commit is because that commit comes first, so all the later commits just piggyback off of it.

**Comment by Sjors** [comment]
@ryanofsky I removed that comment

**Review by ryanofsky — APPROVED** [review]
utACK 8bb3e4c487500a580e3e18791b1f4e7dcdd35442, looks good to merge after the branch

**Comment by practicalswift** [comment]
`GetScriptForMining` is unused after the removal of `generate`.

I suggest getting rid of it! :-)

**Comment by MarcoFalke** [comment]
ACK

**Pull request merged**
